spring-cloud-gcp-data-spanner/src/main/java/org/springframework/cloud/gcp/data/spanner/core/SpannerTemplate.java

```diff
@@ -295,27 +302,53 @@ public T run(TransactionContext transaction) throws Exception {
 
 	private ResultSet executeRead(String tableName, KeySet keys, Iterable<String> columns,
 			SpannerReadOptions options) {
+		StringBuilder logSb = new StringBuilder("Executing read on table " + tableName
```

Comment:
It is best practice to separate the logging logic to make it conditional on `LOGGER.isDebugEnabled()` or similar to avoid performance impact.

Reply:
done